src/nvim/tui/tui.c

```diff
@@ -135,6 +136,7 @@ typedef struct {
   char *space_buf;
 } TUIData;
 
+static bool volatile first_resize = true;
```

Comment:
usual convention is "did" prefix:
```suggestion
static bool volatile did_resize = true;
```

Reply:
Went ahead an renamed, and also moved into scope where I think its purpose clearer, it also didn't need `volatile`.

Semantically `did_user_set_dimensions` is to not resize the **host terminal window**.

This PR and the new variable `did_first_resize` is about resizing the **TUI** correctly and **only at startup**.